main/src/com/google/refine/templating/Parser.java

```diff
@@ -86,6 +92,10 @@ static public Template parse(String s) throws ParsingException {
                     String expression = s.substring(current + 2, closeBrace);
 
                     if (current > start) {
+                    	// Regex to match invalid keys of the template exporter 
+                    	if (!s.substring(start, current).matches("^\\s*,?\\s*\".*\"\\s*:\\s*$")) {
+                    		throw new ParsingException("Missing or bad template");
+                    	}
```

Comment:
I still don't understand what you mean by "keys". There is simply no notion of keys in the templating exporter. I think this `if` statement should be removed entirely.

Reply:
I guess ```StaticFragment``` is a better way, this If statement ensures that the ```StaticFragment``` is valid, a ```StaticFragment``` here which is a ```substring(start, current)```, the valid values can be:
1. ``` { "foo":```
2. ``` { "fo}o":```
3. ```, { "foo":```

This if statement tests for the following invalid values:
1. ``` { {"foo":```
2. ``` { "foo" { :```
3. ``` bar "foo"  : ```

Reply:
The three values that you consider invalid are actually valid.

Reply:
Oh I see, I might have misunderstood the concept, will remove the regex test, then it would add the ```StaticFragment``` as it is. Althought bit curious, if for instance, let choose:
{ ABC"URN": {{jsonize(cells["URN"].value)}} }

This will generate a JSON:
```
{
    "rows" : [
        {
              ABC"URN": 23434
        },
        {
              ABC"URN": 1111
        }
    ]
}
```

which is an incorrect JSON.